cmd/src/debug_kube.go

```diff
@@ -0,0 +1,320 @@
+
+	// create goroutine to get kubectl events
+	run(func() *archiveFile { return getEvents(ctx, namespace, baseDir) })
+
+	// create goroutine to get persistent volumes
+	run(func() *archiveFile { return getPV(ctx, namespace, baseDir) })
+
+	// create goroutine to get persistent volumes claim
+	run(func() *archiveFile { return getPVC(ctx, namespace, baseDir) })
+
+	// start goroutine to run kubectl logs for each pod's container's
+	for _, pod := range pods.Items {
+		for _, container := range pod.Spec.Containers {
+			p := pod.Metadata.Name
+			c := container.Name
+			run(func() *archiveFile { return getPodLog(ctx, p, c, namespace, baseDir) })
+		}
+	}
+
+	// start goroutine to run kubectl logs --previous for each pod's container's
+	// won't write to zip on err, only passes bytes to channel if err not nil
+	for _, pod := range pods.Items {
+		for _, container := range pod.Spec.Containers {
+			p := pod.Metadata.Name
+			c := container.Name
+			run(func() *archiveFile {
+				f := getPastPodLog(ctx, p, c, namespace, baseDir)
+				if f.err != nil {
+					if verbose {
+						fmt.Printf("Could not gather --previous pod logs for %s\n", p)
+					}
+					return nil
+				}
+				return f
+			})
+		}
+	}
+
+	// start goroutine for each pod to run kubectl describe pod
+	for _, pod := range pods.Items {
+		p := pod.Metadata.Name
+		run(func() *archiveFile { return getDescribe(ctx, p, namespace, baseDir) })
+	}
+
+	// start goroutine for each pod to run kubectl get pod <pod> -o yaml
+	for _, pod := range pods.Items {
+		p := pod.Metadata.Name
+		run(func() *archiveFile { return getManifest(ctx, p, namespace, baseDir) })
+	}
+
+	// start goroutine to get external service config
+	if !noConfigs {
+		run(func() *archiveFile { return getExternalServicesConfig(ctx, baseDir) })
+
+		run(func() *archiveFile { return getSiteConfig(ctx, baseDir) })
+	}
+
+	// close channel when wait group goroutines have completed
+	go func() {
+		if err := g.Wait(); err != nil {
+			fmt.Printf("archiveKube failed to open wait group: %s\n", err)
+			os.Exit(1)
+		}
+		close(ch)
+	}()
+
+	// Read binaries from channel and write to archive on host machine
+	if err := writeChannelContentsToZip(zw, ch, verbose); err != nil {
+		return errors.Wrap(err, "failed to write archives from channel")
+	}
+
+	return nil
+}
+
+// Calls kubectl get pods and returns a list of pods to be processed
+func selectPods(ctx context.Context, namespace string) (podList, error) {
+	// Declare buffer type var for kubectl pipe
+	var podsBuff bytes.Buffer
+
+	// Get all pod names as json
+	podsCmd := exec.CommandContext(
+		ctx,
+		"kubectl", "-n", namespace, "get", "pods", "-l", "deploy=sourcegraph", "-o=json",
+	)
+	podsCmd.Stdout = &podsBuff
+	podsCmd.Stderr = os.Stderr
+	err := podsCmd.Run()
+	if err != nil {
+		errors.Wrap(err, "failed to aquire pods for subcommands with err")
+	}
+
+	//Decode json from podList
+	var pods podList
+	if err := json.NewDecoder(&podsBuff).Decode(&pods); err != nil {
```

Comment:
This can be simplified to:

```suggestion
	if err := json.Unmarshal(podsBuff.Bytes(), &pods); err != nil {
```